File: app/src/main/java/com/fitareq/assessmentapphpl/data/api/ApiService.java
package com.fitareq.assessmentapphpl.data.api;

import com.fitareq.assessmentapphpl.data.model.NewTaskBody;
import com.fitareq.assessmentapphpl.data.model.RegistrationBody;
import com.fitareq.assessmentapphpl.data.model.TaskHistory;
import com.fitareq.assessmentapphpl.data.model.TaskHistoryBody;

import java.util.List;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface ApiService {
    @POST("Registration")
    Call<ResponseBody> userRegistration(@Body RegistrationBody body);

    @POST("TaskEntry")
    Call<ResponseBody> addNewTask(@Body NewTaskBody body);

    @POST("TaskData")
    Call<ResponseBody> getHistory(@Body TaskHistoryBody body);
}

File: app/src/main/java/com/fitareq/assessmentapphpl/data/api/ApiService.kt
package com.fitareq.assessmentapphpl.data.api
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface ApiService {
    @POST("login")
    suspend fun userLogin(
    )

    @POST("register")
    suspend fun userRegister(
    )
}